libglusterfs/src/syncop.c

```diff
@@ -474,10 +469,22 @@ synctask_create(struct syncenv *env, size_t stacksize, synctask_fn_t fn,
     if (destroymode)
         return NULL;
 
-    newtask = GF_CALLOC(1, sizeof(*newtask), gf_common_mt_synctask);
+    if (stacksize <= 0) {
+        newtask = GF_MALLOC(sizeof(struct synctask) + env->stacksize,
+                            gf_common_mt_synctask);
+        newtask->ctx.uc_stack.ss_size = env->stacksize;
```

Comment:
This, and line 479, are wrong - need to check (as we do in line 482) that newtask != NULL